DTDL/publication.json

```diff
@@ -0,0 +1,100 @@
+{
+  "@context": "dtmi:dtdl:context;2",
+  "@id": "dtmi:io:tributech:publication;1",
+  "@type": "Interface",
+  "displayName": "Publication",
+  "description": "A Publication represents the will of a Node to share data with the entire hub/ecosystem or selected Nodes.",
+  "contents": [
+    {
+      "@type": "Property",
+      "name": "AvailableFrom",
+      "schema": "dateTime",
+      "writable": true,
+      "displayName": "Available From",
+      "description": "Start date/time for data to be shared (inclusive). (optional)",
+      "comment": "Inclusive means that values starting from inclusive (>=) From will be included in the range."
+    },
+    {
+      "@type": "Property",
+      "name": "AvailableUntil",
+      "schema": "dateTime",
+      "writable": true,
+      "displayName": "Available Until",
+      "description": "End date/time for data to be shared (exclusive). (optional)",
+      "comment": "Exclusive means that only values before (<) To will be included in the range."
+    },
+    {
+      "@type": "Property",
+      "name": "Kind",
+      "schema": {
+        "@type": "Enum",
+        "valueSchema": "string",
+        "enumValues": [
+          {
+            "name": "Public",
+            "displayName": "Public",
+            "enumValue": "PUBLIC"
+          },
+          {
+            "name": "Restricted",
+            "displayName": "Restricted",
+            "enumValue": "RESTRICTED"
+          },
+          {
+            "name": "Disabled",
+            "displayName": "Disabled",
+            "enumValue": "DISABLED"
+          }
+        ]
+      },
+      "writable": true,
+      "displayName": "Kind",
+      "description": "The kind of the Publication."
+    },
+    {
+      "@type": "Property",
+      "name": "PublicationDate",
+      "schema": "dateTime",
+      "writable": true,
+      "displayName": "Publication Date",
+      "description": "The date/time when the Publication was created."
+    },
+    {
+      "@type": "Relationship",
+      "name": "Dataset",
+      "minMultiplicity": 0,
+      "maxMultiplicity": 1,
```

Comment:
Till now it was possible to select the valuemetadataids of the sources of a dataset which should be published. Now all valuemetadataids of the dataset would be published. Was this changed on purpose?

The limitation to a single dataset was only introduced at the former backend due to being a requirement of the UI. For the future this could also be less strict.

Reply:
we should keep a reference to the dataset,but i agree i think we need an additional relationship that maps to specific streams (from the dataset) will be interesting how we enforce consistency there

Reply:
what do you think about extending the dataset?

Reply:
do we even need a dataset at all?

Reply:
At the Metadata-API we only referenced the valuemetadataids (now streams), since you can always traverse from them to source and dataset.